aleph-client/src/elections.rs

```diff
@@ -23,37 +21,48 @@ pub fn get_committee_seats<C: AnyConnection>(
         })
 }
 
-pub fn get_next_era_committee_seats<C: AnyConnection>(connection: &C) -> CommitteeSeats {
+pub fn get_next_era_committee_seats<C: AnyConnectionExt>(connection: &C) -> CommitteeSeats {
     connection.read_storage_value(PALLET, "NextEraCommitteeSize")
 }
 
-pub fn get_validator_block_count<C: AnyConnection>(
+pub fn get_validator_block_count<C: AnyConnectionExt>(
     connection: &C,
     account_id: &AccountId,
     block_hash: Option<H256>,
 ) -> Option<u32> {
-    connection
-        .as_connection()
-        .get_storage_map(PALLET, "SessionValidatorBlockCount", account_id, block_hash)
-        .expect("Failed to obtain SessionValidatorBlockCount extrinsic!")
+    connection.read_storage_map(PALLET, "SessionValidatorBlockCount", account_id, block_hash)
 }
 
-pub fn get_current_era_validators<C: AnyConnection>(connection: &C) -> EraValidators<AccountId> {
+pub fn get_current_era_validators<C: AnyConnectionExt>(connection: &C) -> EraValidators<AccountId> {
     connection.read_storage_value(PALLET, "CurrentEraValidators")
 }
 
-pub fn get_current_era_reserved_validators<C: AnyConnection>(connection: &C) -> Vec<AccountId> {
+pub fn get_current_era_reserved_validators<C: AnyConnectionExt>(connection: &C) -> Vec<AccountId> {
     get_current_era_validators(connection).reserved
 }
 
-pub fn get_current_era_non_reserved_validators<C: AnyConnection>(connection: &C) -> Vec<AccountId> {
+pub fn get_current_era_non_reserved_validators<C: AnyConnectionExt>(
+    connection: &C,
+) -> Vec<AccountId> {
     get_current_era_validators(connection).non_reserved
 }
 
-pub fn get_next_era_reserved_validators<C: AnyConnection>(connection: &C) -> Vec<AccountId> {
+pub fn get_next_era_reserved_validators<C: AnyConnectionExt>(connection: &C) -> Vec<AccountId> {
     connection.read_storage_value(PALLET, "NextEraReservedValidators")
 }
 
-pub fn get_next_era_non_reserved_validators<C: AnyConnection>(connection: &C) -> Vec<AccountId> {
+pub fn get_next_era_non_reserved_validators<C: AnyConnectionExt>(connection: &C) -> Vec<AccountId> {
     connection.read_storage_value(PALLET, "NextEraNonReservedValidators")
 }
+
+pub fn get_era_validators<C: AnyConnectionExt>(
+    connection: &C,
+    session_index: SessionIndex,
+) -> EraValidators<AccountId> {
+    let session_period = get_session_period(connection);
+    let block_number = session_period * session_index;
+    let block_hash = get_block_hash(connection, block_number);
```

Comment:
you can reuse `get_session_first_block()`

Reply:
Changed.